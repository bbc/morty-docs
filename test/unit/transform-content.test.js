const transformContent = require('../../src/transform-content.js')

const options = {
  contentTitle: 'Some Title',
  basePath: '/morty-docs/some-repo'
}

describe('Transform-content returns correct relative path', () => {
  it('for the html made from .md', () => {
    const inputObj = {
      relativePath: 'simple-content.md',
      raw: 'NotUsedInTest-SoFar'
    }

    const result = transformContent(inputObj, options)

    expect(result).toEqual(
      expect.objectContaining({ relativePath: 'simple-content.html', raw: expect.anything() })
    )
  })

  it('even when there is a dot in the path to input .md', () => {
    const inputObj = {
      relativePath: 'docs/arch.2019/simple-content.md',
      raw: 'NotUsedInTest-SoFar'
    }

    const result = transformContent(inputObj, options)

    expect(result).toEqual(
      expect.objectContaining({ relativePath: 'docs/arch.2019/simple-content.html', raw: expect.anything() })
    )
  })
})

// TODO: ? we could extract different parts of the the html and test them separately
describe('Transform-content returns the correct html from markdown', () => {
  it('for a line of plain text', () => {
    const inputObj = {
      relativePath: 'file-name.md',
      raw: 'One line of plain text'
    }

    const result = transformContent(inputObj, options)

    expect(result.raw).toMatchSnapshot()
  })

  it('for Markdown with a title and links', () => {
    const inputObj = {
      relativePath: 'file-name.md',
      raw: '# Title of simple-content.md \n[Relative link to MD in docs directory](docs/file-to-publish.md)\n[Root link to MD in docs directory](/docs/file-to-publish.md)'
    }

    const result = transformContent(inputObj, options)

    expect(result.raw).toMatchSnapshot()
  })
})

describe('Transform-content returns the correct html from AsciiDoc', () => {
  it('for a line of plain text', () => {
    const inputObj = {
      relativePath: 'file-name.asciidoc',
      raw: 'One line of plain text'
    }

    const result = transformContent(inputObj, options)

    expect(result.raw).toMatchSnapshot()
  })

  it('for .asciidoc with a title and a link', () => {
    const inputObj = {
      relativePath: 'file-name.asciidoc',
      raw: '== Title of simple-content.asciidoc \nlink:docs/file-to-publish.md[Link to MD in docs directory]'
    }

    const result = transformContent(inputObj, options)

    expect(result.raw).toMatchSnapshot()
  })

  it('for .adoc with a title and a link', () => {
    const inputObj = {
      relativePath: 'file-name.adoc',
      raw: '== Title of simple-content.adoc \nlink:docs/file-to-publish.md[Link to MD in docs directory]'
    }

    const result = transformContent(inputObj, options)

    expect(result.raw).toMatchSnapshot()
  })

  it('for .asc with a title and a link', () => {
    const inputObj = {
      relativePath: 'file-name.asc',
      raw: '== Title of simple-content.asc \nlink:docs/file-to-publish.md[Link to MD in docs directory]'
    }

    const result = transformContent(inputObj, options)

    expect(result.raw).toMatchSnapshot()
  })
})
