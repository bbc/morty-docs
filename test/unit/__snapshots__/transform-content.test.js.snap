// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transform-content returns the correct html from AsciiDoc for .adoc with a title and a link 1`] = `
<html lang="en"
      style="min-height:100vh"
      data-reactroot
>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <title>
      file-name.adoc
    </title>
    <link rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
          integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
          crossorigin="anonymous"
    >
  </head>
  <body style="min-height:100vh">
    <div style="min-height:75vh;padding-bottom:20px">
      <div class="container"
           style="border:none;border-radius:0;background-color:#000;height:60px;margin-bottom:0;width:100%;padding:1.142rem"
      >
        <nav>
          <ol style="padding:0;list-style:none;display:flex;height:100%;align-items:center">
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs"
              >
                morty-docs
              </a>
            </li>
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs/some-repo"
              >
                some-repo
              </a>
            </li>
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs/some-repo/file-name.adoc"
              >
                file-name.adoc
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div class="container">
        <div class="sect1">
          <h2 id="_title_of_simple_content_adoc">
            Title of simple-content.adoc
          </h2>
          <div class="sectionbody">
            <div class="paragraph">
              <p>
                <a href="docs/file-to-publish.md">
                  Link to MD in docs directory
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer style="bottom:0;width:100%;background-color:#f5f5f5;padding:0 15px;box-sizing:border-box;min-height:25vh;position:relative"
            class="footer"
    >
      <div style="padding:1em 0 2em 0"
           class="container"
      >
        <div class="row">
          <div style="text-align:center"
               class="col-md-4 col-md-offset-4"
          >
            <h3>
              What is Morty-Docs?
            </h3>
            <p>
              Link to
              <a href="https://github.com/bbc/morty-docs">
                Morty-docs
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </footer>
  </body>
</html>
`;

exports[`Transform-content returns the correct html from AsciiDoc for .asc with a title and a link 1`] = `
<html lang="en"
      style="min-height:100vh"
      data-reactroot
>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <title>
      file-name.asc
    </title>
    <link rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
          integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
          crossorigin="anonymous"
    >
  </head>
  <body style="min-height:100vh">
    <div style="min-height:75vh;padding-bottom:20px">
      <div class="container"
           style="border:none;border-radius:0;background-color:#000;height:60px;margin-bottom:0;width:100%;padding:1.142rem"
      >
        <nav>
          <ol style="padding:0;list-style:none;display:flex;height:100%;align-items:center">
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs"
              >
                morty-docs
              </a>
            </li>
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs/some-repo"
              >
                some-repo
              </a>
            </li>
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs/some-repo/file-name.asc"
              >
                file-name.asc
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div class="container">
        <div class="sect1">
          <h2 id="_title_of_simple_content_asc">
            Title of simple-content.asc
          </h2>
          <div class="sectionbody">
            <div class="paragraph">
              <p>
                <a href="docs/file-to-publish.md">
                  Link to MD in docs directory
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer style="bottom:0;width:100%;background-color:#f5f5f5;padding:0 15px;box-sizing:border-box;min-height:25vh;position:relative"
            class="footer"
    >
      <div style="padding:1em 0 2em 0"
           class="container"
      >
        <div class="row">
          <div style="text-align:center"
               class="col-md-4 col-md-offset-4"
          >
            <h3>
              What is Morty-Docs?
            </h3>
            <p>
              Link to
              <a href="https://github.com/bbc/morty-docs">
                Morty-docs
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </footer>
  </body>
</html>
`;

exports[`Transform-content returns the correct html from AsciiDoc for .asciidoc with a title and a link 1`] = `
<html lang="en"
      style="min-height:100vh"
      data-reactroot
>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <title>
      file-name.asciidoc
    </title>
    <link rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
          integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
          crossorigin="anonymous"
    >
  </head>
  <body style="min-height:100vh">
    <div style="min-height:75vh;padding-bottom:20px">
      <div class="container"
           style="border:none;border-radius:0;background-color:#000;height:60px;margin-bottom:0;width:100%;padding:1.142rem"
      >
        <nav>
          <ol style="padding:0;list-style:none;display:flex;height:100%;align-items:center">
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs"
              >
                morty-docs
              </a>
            </li>
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs/some-repo"
              >
                some-repo
              </a>
            </li>
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs/some-repo/file-name.asciidoc"
              >
                file-name.asciidoc
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div class="container">
        <div class="sect1">
          <h2 id="_title_of_simple_content_asciidoc">
            Title of simple-content.asciidoc
          </h2>
          <div class="sectionbody">
            <div class="paragraph">
              <p>
                <a href="docs/file-to-publish.md">
                  Link to MD in docs directory
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer style="bottom:0;width:100%;background-color:#f5f5f5;padding:0 15px;box-sizing:border-box;min-height:25vh;position:relative"
            class="footer"
    >
      <div style="padding:1em 0 2em 0"
           class="container"
      >
        <div class="row">
          <div style="text-align:center"
               class="col-md-4 col-md-offset-4"
          >
            <h3>
              What is Morty-Docs?
            </h3>
            <p>
              Link to
              <a href="https://github.com/bbc/morty-docs">
                Morty-docs
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </footer>
  </body>
</html>
`;

exports[`Transform-content returns the correct html from AsciiDoc for a line of plain text 1`] = `
<html lang="en"
      style="min-height:100vh"
      data-reactroot
>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <title>
      file-name.asciidoc
    </title>
    <link rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
          integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
          crossorigin="anonymous"
    >
  </head>
  <body style="min-height:100vh">
    <div style="min-height:75vh;padding-bottom:20px">
      <div class="container"
           style="border:none;border-radius:0;background-color:#000;height:60px;margin-bottom:0;width:100%;padding:1.142rem"
      >
        <nav>
          <ol style="padding:0;list-style:none;display:flex;height:100%;align-items:center">
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs"
              >
                morty-docs
              </a>
            </li>
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs/some-repo"
              >
                some-repo
              </a>
            </li>
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs/some-repo/file-name.asciidoc"
              >
                file-name.asciidoc
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div class="container">
        <div class="paragraph">
          <p>
            One line of plain text
          </p>
        </div>
      </div>
    </div>
    <footer style="bottom:0;width:100%;background-color:#f5f5f5;padding:0 15px;box-sizing:border-box;min-height:25vh;position:relative"
            class="footer"
    >
      <div style="padding:1em 0 2em 0"
           class="container"
      >
        <div class="row">
          <div style="text-align:center"
               class="col-md-4 col-md-offset-4"
          >
            <h3>
              What is Morty-Docs?
            </h3>
            <p>
              Link to
              <a href="https://github.com/bbc/morty-docs">
                Morty-docs
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </footer>
  </body>
</html>
`;

exports[`Transform-content returns the correct html from markdown for Markdown with a title and links 1`] = `
<html lang="en"
      style="min-height:100vh"
      data-reactroot
>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <title>
      file-name.md
    </title>
    <link rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
          integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
          crossorigin="anonymous"
    >
  </head>
  <body style="min-height:100vh">
    <div style="min-height:75vh;padding-bottom:20px">
      <div class="container"
           style="border:none;border-radius:0;background-color:#000;height:60px;margin-bottom:0;width:100%;padding:1.142rem"
      >
        <nav>
          <ol style="padding:0;list-style:none;display:flex;height:100%;align-items:center">
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs"
              >
                morty-docs
              </a>
            </li>
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs/some-repo"
              >
                some-repo
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div class="container">
        <h1 id="title-of-simple-contentmd">
          <a href="#title-of-simple-contentmd">
            Title of simple-content.md
          </a>
        </h1>
        <p>
          <a href="docs/file-to-publish.html">
            Relative link to MD in docs directory
          </a>
          <br>
          <a href="/morty-docs/some-repo/docs/file-to-publish.html">
            Root link to MD in docs directory
          </a>
        </p>
      </div>
    </div>
    <footer style="bottom:0;width:100%;background-color:#f5f5f5;padding:0 15px;box-sizing:border-box;min-height:25vh;position:relative"
            class="footer"
    >
      <div style="padding:1em 0 2em 0"
           class="container"
      >
        <div class="row">
          <div style="text-align:center"
               class="col-md-4 col-md-offset-4"
          >
            <h3>
              What is Morty-Docs?
            </h3>
            <p>
              Link to
              <a href="https://github.com/bbc/morty-docs">
                Morty-docs
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </footer>
  </body>
</html>
`;

exports[`Transform-content returns the correct html from markdown for a line of plain text 1`] = `
<html lang="en"
      style="min-height:100vh"
      data-reactroot
>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <title>
      file-name.md
    </title>
    <link rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
          integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
          crossorigin="anonymous"
    >
  </head>
  <body style="min-height:100vh">
    <div style="min-height:75vh;padding-bottom:20px">
      <div class="container"
           style="border:none;border-radius:0;background-color:#000;height:60px;margin-bottom:0;width:100%;padding:1.142rem"
      >
        <nav>
          <ol style="padding:0;list-style:none;display:flex;height:100%;align-items:center">
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs"
              >
                morty-docs
              </a>
            </li>
            <li>
              <span aria-hidden="true"
                    style="margin:0 0.28rem;color:#ebebeb"
              >
                /
              </span>
              <a style="text-align:left;color:lightblue;font-size:2rem"
                 href="/morty-docs/some-repo"
              >
                some-repo
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div class="container">
        <p>
          One line of plain text
        </p>
      </div>
    </div>
    <footer style="bottom:0;width:100%;background-color:#f5f5f5;padding:0 15px;box-sizing:border-box;min-height:25vh;position:relative"
            class="footer"
    >
      <div style="padding:1em 0 2em 0"
           class="container"
      >
        <div class="row">
          <div style="text-align:center"
               class="col-md-4 col-md-offset-4"
          >
            <h3>
              What is Morty-Docs?
            </h3>
            <p>
              Link to
              <a href="https://github.com/bbc/morty-docs">
                Morty-docs
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </footer>
  </body>
</html>
`;
